### ISSUE TITLE
[Feature Request]: RUSTC_AR is missing

### ISSUE THREAD

**Issue by alexisart** [rust-lang/cargo#12183]
### Problem

RUSTC_AR is missing from the available build environment variables in build.rs.

This makes the below code not work.

```rust
    cc::Build::new()
        .compiler(PathBuf::from(env::var("RUSTC_LINKER").unwrap()))
        .archiver(PathBuf::from(env::var("RUSTC_AR").unwrap()))
        .file(lib_path.to_str().unwrap())
        .compile("browser");
```

### Proposed Solution

I started work on https://github.com/alexisart/cargo/commit/287fb2aceb0ba5e4d427b1af0fe7e97bec17f283 and am currently compiling Cargo in order to make sure the variable works. 4 tests fail, so I temporarily disabled them. The 4 tests are listed below.

```bash
cargo test -j 8 build_script::custom_build_ar_bad_host
cargo test -j 8 build_script::custom_build_ar_bad_host_with_arch
cargo test -j 8 build_script::custom_build_ar_host_target_with_bad_host_config
cargo test -j 8 build_script::custom_build_env_var_rustc_ar_bad_host_target
```

### Notes

I based my commit on https://github.com/rust-lang/cargo/pull/5394 and am waiting for approval to make a pull request to add this feature

**Comment by weihanglo** [comment]
Hello. As far as I can tell

* The codegen option in rustc (`-C ar=`) [has been deprecated](https://doc.rust-lang.org/rustc/codegen-options/index.html#ar) long time ago.
* rustc switched to `llvm-ar` instead of external `ar` since 2015 — <https://github.com/rust-lang/rust/pull/26926>
* `target.<triple>.ar` [is not used anymore in cargo](https://doc.rust-lang.org/nightly/cargo/reference/config.html#targettriplear) — https://github.com/rust-lang/cargo/pull/2369

Basically it does nothing on rustc/cargo side. I am not sure if this configuration is still useful. Could you elaborate more?

**Comment by alexisart** [comment]
That's good to know. I've had my config.toml setup as:

```toml
[target.wasm32-unknown-emscripten]
ar = "$WORKSPACE/tools/emsdk/upstream/emscripten/emar"
linker = "$WORKSPACE/tools/emsdk/upstream/emscripten/emcc"
rustflags = ["-L$WORKSPACE/target/libs", "-lSDL2", "-lSDL2_image", "-lSDL2_ttf", "-lSDL2_mixer", "-lpthread"]
```

The $WORKSPACE variable is replaced by sed in my build script. I was using the ar variable thinking it was telling rust to use emar instead of the default ar tool.

**Comment by weihanglo** [comment]
Thank you for letting us know.

Unless people are using an extremely old Rust, this should be a no-op. Going in close this out as we don't support archiver configuration anymore. Thanks.
